disnake/embeds.py

```diff
@@ -268,20 +300,31 @@ def copy(self) -> Self:
         embed._files = self._files.copy()
         return embed
 
-    def __len__(self) -> int:
-        total = len(self.title or "") + len(self.description or "")
-        if self._fields:
-            for field in self._fields:
-                total += len(field["name"]) + len(field["value"])
+    @staticmethod
+    def __len(title: Optional[str], description: Optional[str],
+              fields: Optional[List[EmbedFieldPayload]] = None, footer: Optional[EmbedFooterPayload] = None,
+              author: Optional[EmbedAuthorPayload] = None, strip: bool = True):
+
+        def stripped_length(text: str) -> int:
+            return len(text.strip() if strip else text)
+
+        total = stripped_length(title or "") + stripped_length(description or "")
+        if fields:
+            for field in fields:
+                total += stripped_length(field["name"]) + stripped_length(field["value"])
 
-        if self._footer and (footer_text := self._footer.get("text")):
-            total += len(footer_text)
+        if footer and (footer_text := footer.get("text")):
+            total += stripped_length(footer_text)
 
-        if self._author and (author_name := self._author.get("name")):
-            total += len(author_name)
+        if author and (author_name := author.get("name")):
+            total += stripped_length(author_name)
 
         return total
 
+    def __len__(self) -> int:
```

Comment:
Should the len always be the stripped calculation?

Reply:
I think that discord does their calculations for length limits after stripping, but we'd need to double check if they guarantee this.

Reply:
This is pretty certain, we can go with the stripped values.